File: Java_MVP/app/src/main/java/com/gotoinc/requesin/view/home/HomeActivity.java
package com.gotoinc.requesin.view.home;

import android.os.Bundle;

import com.google.android.material.snackbar.Snackbar;
import com.gotoinc.requesin.R;
import com.gotoinc.requesin.view.BaseViewContract;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;

public class HomeActivity extends AppCompatActivity implements BaseViewContract {
    private ConstraintLayout root;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        findViews();
        showFirstFragment();
    }

    private void findViews() {
        root = findViewById(R.id.root);
    }

    private void showFirstFragment() {
        getSupportFragmentManager().beginTransaction()
                .add(R.id.container, HomeFragment.getInstance(), HomeFragment.class.getSimpleName())
                .commit();
    }

    @Override
    public void onError() {
        Snackbar.make(root, R.string.error_unknown, Snackbar.LENGTH_SHORT).show();
    }

    @Override
    public void onError(String message) {
        Snackbar.make(root, message, Snackbar.LENGTH_SHORT).show();
    }

    @Override
    public void onTimeout() {
        Snackbar.make(root, R.string.error_unknown, Snackbar.LENGTH_SHORT).show();
    }

    @Override
    public void onNetworkError() {
        Snackbar.make(root, R.string.error_unknown, Snackbar.LENGTH_SHORT).show();
    }
}

File: Kotlin_MVP/app/src/main/java/com/gotoinc/requesin/view/home/HomeActivity.kt
package com.gotoinc.requesin.view.home

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar
import com.gotoinc.requesin.R
import com.gotoinc.requesin.view.BaseViewContract
import kotlinx.android.synthetic.main.activity_home.*

class HomeActivity : AppCompatActivity(), BaseViewContract {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        showFirstFragment()
    }

    private fun showFirstFragment() {
        supportFragmentManager.beginTransaction()
            .add(R.id.container, HomeFragment.getInstance(), HomeFragment.TAG)
            .commit()
    }

    override fun onError() {
        Snackbar.make(root_layout, R.string.error_unknown, Snackbar.LENGTH_SHORT).show()
    }

    override fun onError(message: String) {
        Snackbar.make(root_layout, message, Snackbar.LENGTH_SHORT).show()
    }

    override fun onTimeout() {
        Snackbar.make(root_layout, R.string.error_unknown, Snackbar.LENGTH_SHORT).show()
    }

    override fun onNetworkError() {
        Snackbar.make(root_layout, R.string.error_unknown, Snackbar.LENGTH_SHORT).show()
    }
}
